kernel/src/table_configuration.rs

```diff
@@ -0,0 +1,275 @@
+//! This module defines [`TableConfiguration`], a high level api to check feature support/enablement for a table.
+//! This encapsulates [`Protocol`], [`Metadata`], and extracts the [`TableProperties`] and [`ColumnMappingMode`].
+//! Protocol and Metadata are deeply intertwined when dealing with table features and
```

Comment:
maybe add that it effectively consumes protocol/metadata but encapsulates all of metadata/protocol/schema/properties/features/etc.